src/celeritas/optical/ScintillationParams.cc

```diff
@@ -0,0 +1,31 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/ScintillationParams.cc
+//---------------------------------------------------------------------------//
+#include "ScintillationParams.hh"
+
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "celeritas/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with scintillation data.
+ */
+ScintillationParams::ScintillationParams(ScintillationDataCRef const& data)
+{
+    CELER_EXPECT(data);
+    HostVal<ScintillationData> host_data;
```

Comment:
Where should we put error checking (positive mean/sigma for the scintillation components, yields add up to 1, etc)?

Reply:
This may be done when we build Params from the process level - and we can revisit later as necessary.